tests/test_ic8report.py

```diff
@@ -0,0 +1,55 @@
+import os
+from typing import Dict
+
+import pandas as pd
+import pathlib
+
+path_to_data_for_tests = (
+    pathlib.Path(os.path.dirname(__file__)) / "code_and_data_for_tests"
+)
+
+def run_ic8_report(tmpdir: pathlib.Path) -> Dict:
+    """Returns the results generated by the running the Report class. (Also generates that Excel file in a temporary
+    location.)"""
+
+    from scripts.ic8.analyses.main_results_for_investment_case import get_report
+
+    r = get_report(
+        load_data_from_raw_files=True,
+        do_checks=False,
+        run_analysis=True,
+    )
+
+    # Generate report (checking that it can be written to an Excel file)
+    filename = tmpdir / 'test_report_ic8.xlsx'
+    rtn_from_return = r.report(filename)
+
+    # Return the results of the report
+    return rtn_from_return
+
+"""
+Warning to ensure this runs make sure in the parameters.toml you choose the full list of countries for portfolio and 
+modelled countries, otherwise this code will throw and error. 
+"""
+
+def test_ic8report(tmpdir):
+    """
+    This test runs the report for IC8 (including loading the model results from scratch and running the optimisation
+     analysis) and checks that the results are the same as that obtained previously and which have been agreed to be
```

Comment:
comment on this line says "all checks to be done", but actually the checks are not called: so recommend editing the text.